### ISSUE TITLE
swan error when running inlet_test

### ISSUE THREAD

**Issue by icebearhuang** [DOI-USGS/COAWST#2]
Hi:

I have successfully compiled the inlet_test case in COAWST. But when I'm running it gives the following error:

`
At line 584 of file swanmain.f90
Fortran runtime error: Allocatable actual argument 'aicef' is not allocated

#0  0x3003c25 in swan_run_
        at /gpfs/projects/WilsonGroup/COAWST/Build/swanmain.f90:584
#1  0x2ea5bd7 in __waves_control_mod_MOD_swan_driver_init
        at /gpfs/projects/WilsonGroup/COAWST/Build/waves_control.f90:93
#2  0x40f5c3 in mct_driver
        at /gpfs/projects/WilsonGroup/COAWST/Build/master.f90:214
#3  0x40fcae in main
        at /gpfs/projects/WilsonGroup/COAWST/Build/master.f90:20
`

AICEF is the ice coverage, but there is no ice module activated in the swan input. Could anyone give some suggestions in solving this problem? 

Thanks

**Comment by jcwarner-usgs** [comment]
ok.  I downloaded the code, compiled, and ran inlet_test on my desktop with gfortran and did the same procedure on the cluster with ifort and they both worked. But .. I think i know the issue.  It might be compiler dependent.  The problem was that the vars aicef and hicef were not allocated, then the routine snexti was sending an unallocated array.  but the code has been like this for years, so the problem is a new compiler flag (i believe).  To correct this, I just made some changes to the code to allocate and point to these aicef and hicef arrays.  it  worked for me but i did not have the original problem.
Can you do an update and see if it is fixed now? thanks, -j.

**Comment by icebearhuang** [comment]
Many thanks for your comment and updating, I have updated code, but a new error shoots up:

`At line 370 of file waves_coupler.f90
Fortran runtime error: Index '1' of dimension 1 of array 'globalsegmap_g' above upper bound of 0

Error termination. Backtrace::
#0  0x2ec2490 in __waves_coupler_mod_MOD_initialize_wav_routers
        at /gpfs/projects/WilsonGroup/COAWST-master/Build/waves_coupler.f90:370
#1  0x2ea5b2d in __waves_control_mod_MOD_swan_driver_init
        at /gpfs/projects/WilsonGroup/COAWST-master/Build/waves_control.f90:84
#2  0x40f5c3 in mct_driver
        at /gpfs/projects/WilsonGroup/COAWST-master/Build/master.f90:214
#3  0x40fcae in main
        at /gpfs/projects/WilsonGroup/COAWST-master/Build/master.f90:20`

 But it is weird that there wasn't such problem for the last version, since it seems happens before last mistake. It seems that SWAN have problem read in the grid this time, I have checked my grid address, it has no problem.

**Comment by jcwarner-usgs** [comment]
can you send the full output?  there may have been an error before this.

**Comment by icebearhuang** [comment]
Here is the full output for your reference.

[run.log](https://github.com/jcwarner-usgs/COAWST/files/4800511/run.log)

**Comment by icebearhuang** [comment]
I also kept the previous version (before you change the aicef allocation), and the full output from that version is in this file.

[previous_run.log](https://github.com/jcwarner-usgs/COAWST/files/4800566/previous_run.log)

**Comment by jcwarner-usgs** [comment]
do me a favor and try to run it without debug on.  i think those compiler flags are rather strict and i have seen SWAN code have some issues with strict compiler rules.

**Comment by icebearhuang** [comment]
Thank you very much for your help, I can now run the inlet_test now, both for Rutgers-ROMS and COAWST. 

In my case the problem seems related relating to two separate issue for both platform. One as you pointed out is the strict compiler rules, when I removed the flags '-fsanitize=address -fsanitize=undefined -finit-real=nan -ffpe-trap=invalid,zero,overflow -finit-real=nan -ffpe-trap=invalid,zero,overflow', I can get both Rutgers-ROMS and COAWST run, but stopped at the same file reading issue for SWAN. With these flags, the Rutgers-ROMS cannot even get the SWAN started, but COAWST seems okay. Leaving '-g -fbounds-check -fbacktrace -fcheck=all' on for bug line tracing is okay. 
The other issue I found in my case is related to the 'TOO LONG' directory name for input files for SWAN and MCT. I moved those files to current folder and direct to them by './files', the problem is then solved. I believe this is another 'flag' problem.
